streaming/src/test/scala/org/apache/spark/streaming/UISeleniumSuite.scala

```diff
@@ -162,7 +163,7 @@ class UISeleniumSuite
         outputOpIds.map(_.text) should be (List("0", "1"))
 
         // Check job ids
-        val jobIdCells = findAll(cssSelector( """#batch-job-table a""")).toSeq
+        val jobIdCells = findAll(cssSelector( """#jobId""")).toSeq
```

Comment:
Is it valid to have duplicate ids in HTML elements? really I was just suggesting you filter the results to only consider numbers. Not entirely robust but maybe sufficient.

Reply:
since here we are testing for jobids only, may be its better to add unique tag to identify them.

Reply:
You can add a class maybe, but, rather than even modify the user-facing HTML, I think a little 'hack' here to make it more robust is fine. It's not worth much more effort.

Reply:
added "class" instead of "id" for the Job Ids. Please retest.

Reply:
As I say, I am not sure it makes sense to modify the user facing HTML just for this. Just ignore the "?" here in the test.

Reply:
Ok, i understand. Now, i have removed all the user facing changes and added a filter to filter out the jobIds. I guess this will be fine.